edx_proctoring/views.py

```diff
@@ -446,7 +447,10 @@ def get(self, request):
             )
 
             if usage_key not in user_outline.accessible_sequences:
-                onboarding_exams.remove(onboarding_exam)
+                inaccessible_onboarding_exams.append(onboarding_exam)
```

Comment:
Can you explain why this change fixes the bug?

Reply:
The original code is modifying a list that it is concurrently iterating over, resulting in an incorrect calculation for `onboarding_exams`.

For example, if you have a list `onboarding_exams` with a length of two, and you hit line 449 and remove an exam, the iteration stops short of the second onboarding exam.

Reply:
Got it. Good catch.